docs/docsite/rst/dev_guidelines.rst

```diff
@@ -1048,3 +1048,57 @@ Where one of these reasons apply you should open a pull request proposing the mi
 Unsupported integration tests will not be automatically run by CI.  However, the
 necessary policies should be available so that the tests can be manually run by
 someone performing a PR review or writing a patch.
+
+Unit-tests for AWS plugins
+==========================
+
+Why do we need unit-tests when we've got functional tests
+---------------------------------------------------------
+
+Unit-tests are much faster and more suitable to test corner cases. They also don't depend on a third party service
+and thus, a failure is less likley to be a false positive.
+
+
+How to keep my code simple?
+---------------------------
+
+Ideally, you should break up your code in tiny functions. Each function should have a limited number of parameters
+and a low amount of cross dependencies with the rest of the code (low coupling):
+
+- Don't pass a large data structure to a function if it only use one field. This clarify the inputs of your
+  function (the contract) and also reduce the risk of an unexpected transformation of the data structure
+  from within the function.
+- The boto client object is complex and can be source of unwanted side-effect. It's better to isolate the calls
+  in dedicated functions. These functions will have their own unit-tests.
+- Don't pass the ``module`` object when you only need the read a couple of parameters from ``module.params``.
+  Pass the parameter directly to your function. By doing so, you're explicite about what the function inputs
+  (the contract) and you reduce potential side-effect.
+
+Unit-tests guidlines
+--------------------
+
+Ideally, all the ``module_utils`` should be covered by unit-tests. However we acknoledge that writing unit-tests may
+be challenging and we also accept contribution with no coverage. The test coverage of the other type of plugins is recommand.
+
+- Our tests are run with ``pytest`` and we use the features it provides such as Fixtures, Parametrization.
```

Comment:
Wouldn't make sense to provide a generic example for Fixtures and Parametrization?

Reply:
Pytest documentation is pretty good and cover these features. But we can add some links pointing on was we identify as our best example of well written unit-tests.
This being said, I would suggest to do this in another patch.

Reply:
The pytest docs are very good once you understand what you're looking at, but I found some of the examples a little too abstracted from reality to understand at first especially as it applies to our specific modules and module_utils.